exporter-helper.php

```diff
@@ -1,10 +1,101 @@
 <?php
 /**
- * Post Exporter Helper Function
+ * Campaign Exporter Helper Functions
  *
  * @package P4MT
  */
 
+/**
+ * Generate a bunch of placeholders for use in an IN query.
+ * Such a shame WordPress doesn't offer a decent way to do bind IN statement params and we have to do this ourselves.
+ *
+ * @param array $items The items to generate placeholders for.
+ * @param int   $start_index The start index to use for creating the placeholders.
+ * @return string The generated placeholders string.
+ */
+function generate_list_placeholders( array $items, int $start_index ): string {
+	$placeholders = [];
+	foreach ( range( $start_index, count( $items ) + $start_index - 1 ) as $i ) {
+		$placeholders[] = "%$i\$d";
+	}
+	return implode( ',', $placeholders );
+}
+
+/**
+ * Parse the post content and return all attachment ids used in blocks.
+ *
+ * @param string $content The content to parse.
+ * @return array All attachments used in the blocks.
+ */
+function get_attachments_used_in_content( string $content ): array {
+	$blocks = parse_blocks( $content );
+
+	$attachment_ids = [];
+
+	foreach ( $blocks as $block ) {
```

Comment:
I just found out something using the Inspections PhpStorm plugin: if a foreach contains only a switch statement, you can combine both and only have one set of square brackets instead of two.
`foreach ( $blocks as $block ) switch ( $block['blockName']) {
			case 'planet4-blocks/enform':
				$attachment_ids[] = $block['attrs']['background'] ?? '';
				break;

			case 'core/image':
    // ...
}

Reply:
The nice thing about it is that from the first line you can see the foreach only has a switch in it, whereas otherwise you would need to scroll past the switch to see that there is nothing more in the function.

Reply:
It looks more cleaner way, but `phpcs` won't like it :(